Update da_read_obs_bufrseviri.inc to fix meteosat 11 error messages

WQadam opened the pull request:
TYPE: Bug fix

DESCRIPTION OF CHANGES:
Problem:
Meteosat 11 was added to the GFS bufr files in Feb 2018 meaning when the seviri bufr read was turned on in the namelist it would throw out an error message for each data point in the log file making it very large. The change to the error loop removes this message but also prepares the bufr read file to ingest meteosat 11 data into wrf. Dont think a crtm currently supports meteosat 11 but i can change the VARBC.in file to include data for this sat number and add a msg-4-seviri.info file to the radiance_info folder in another commit if this might be desirable for people to use.

Solution:
Added identifier for meteosat 11 and then changed the upper limit of the identifier error if with the new identifier significantly higher than previous

RELEASE NOTE: Updated bufrseviri.inc so that if seviri turned on in WRFDA after Feb 2018 it suppresses many error messages in log file.

liujake commented:
Thanks for adding this fix! You need to change the head number from 42 to 43 when adding a new satellite in VARBC.in. Please also add a list of change files in the PR message.

WQadam commented:
Not a problem. I have now changed the VARBC.in number of instruments to 43 and updated the original PR comment to add which files i have changed.

weiwangncar commented:
The reg tests have passed:
    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           58
    Number of Simulations  : 158           156        0
    Number of Comparisons  : 95           92        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

thomasuhoh commented:
I think one needs to add 
        _if (satellite_id == 4) crtm_sat='m11'_

after line 35 of da_crtm_sensor_descriptor.inc. Otherwise M11 is not recognized by CRTM.

WQadam commented:
Good spot Thomas. I have updated the da_crtm_sensor_descriptor.inc file with m11 and updated the PR above. 
Looking in that descriptor file i notice the hirs4 and avhrr4 section doesnt have a satellite descriptor for NOAA 19 but the user guide includes NOAA 19. Should lines 51 and 55 the crtm sensor descriptor be"satellite_id >= 18" rather than "=="?
The WRFDA user guide will need to be updated to include meteosat 11 but not sure where that is stored.

WQadam commented:
I have done some tests with just Noaa-19 satelitte activated for AMSUA and hirs4 instrument. To get hirs4 into the log files and activated i needed to add a section to the VARBC.in file as noaa19 hirs4 isnt currently in the VARBC.in file. 
I found subtle differences between the tests i did with one missing noaa19 in the CRTM descriptor file but the other had the changes i made. Both tests ran smoothly and both seemed to suggest in the stats files and log files they were reading and assimilating the hirs4 noaa 19 instrument. 
Posses 2 questions
Could the differences i see in the output be due to the nature of 3dvar and the second run just being an ensemble member or is it due to the crtm now using hirs4 data from noaa-19?
If its due to the crtm now detecting the hirs4 data from noaa-19 then quite a few satellites and sensors are missing from the CRTM sensor descriptor file so might need adding.

liujake commented:
Will get back to this next week.

weiwangncar commented:
The Jenkins tests have passed:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           58
    Number of Simulations  : 158           156        0
    Number of Comparisons  : 95           92        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

liujake commented:
@weiwangncar can you give another approval, then merge?

weiwangncar reviewed (approved):
Per Jake's request.

smileMchen commented:
@WQadam 
Would you please let me know your name and affiliation? Thanks.

WQadam commented:
> @WQadam Would you please let me know your name and affiliation? Thanks.

Adam D
WeatherQuest

Thanks
